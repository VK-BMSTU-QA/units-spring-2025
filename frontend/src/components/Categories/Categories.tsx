import React, { FC } from 'react';
import cn from 'classnames';

import type { Category } from '../../types';

import './Categories.css';

type CategoriesProps = {
    selectedCategories: Category[];
    onCategoryClick?: (category: Category) => void;
};

const allFilters: Category[] = ['Одежда', 'Для дома', 'Электроника'];

export const Categories: FC<CategoriesProps> = ({
    selectedCategories,
    onCategoryClick,
}) => (
    <div className="categories">
        {allFilters.map((filter) => (
            <div
                data-testid={filter}
                key={filter}
                onClick={() => onCategoryClick?.(filter)}
                className={cn('categories__badge', {
                    categories__badge_selected:
                        selectedCategories.includes(filter),
                })}
            >
                {filter}
            </div>
        ))}
    </div>
);
