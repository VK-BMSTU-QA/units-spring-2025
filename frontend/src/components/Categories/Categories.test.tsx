import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Categories } from './Categories';

afterEach(jest.clearAllMocks);

describe('Categories test', () => {
    it('should render correctly', () => {
        const rendered = render(<Categories selectedCategories={[]} />);

        expect(rendered.asFragment()).toMatchSnapshot();
    });

    it('should add class for selected badge', () => {
        const rendered = render(<Categories selectedCategories={['Одежда']} />);

        expect(rendered.getByText('Одежда')).toHaveClass(
            'categories__badge_selected'
        );
        expect(rendered.getByText('Электроника')).not.toHaveClass(
            'categories__badge_selected'
        );
        expect(rendered.getByText('Для дома')).not.toHaveClass(
            'categories__badge_selected'
        );
    });

    it('should call callback when category click', () => {
        const onCategoryClick = jest.fn();
        const rendered = render(
            <Categories
                selectedCategories={[]}
                onCategoryClick={onCategoryClick}
            />
        );

        expect(onCategoryClick).toHaveBeenCalledTimes(0);
        fireEvent.click(rendered.getByText('Одежда'));
        expect(onCategoryClick).toHaveBeenCalledTimes(1);
    });
});
