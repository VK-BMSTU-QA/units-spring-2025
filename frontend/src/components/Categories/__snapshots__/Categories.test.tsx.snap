// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories test should render correctly 1`] = `
<DocumentFragment>
  <div
    class="categories"
  >
    <div
      class="categories__badge"
    >
      Одежда
    </div>
    <div
      class="categories__badge"
    >
      Для дома
    </div>
    <div
      class="categories__badge"
    >
      Электроника
    </div>
  </div>
</DocumentFragment>
`;
